autonat.go

```diff
@@ -323,7 +339,17 @@ func (as *AmbientAutoNAT) probe(pi *peer.AddrInfo) {
 	}
 }
 
-func (as *AmbientAutoNAT) probeNextPeer() {
+func (as *AmbientAutoNAT) probeNextPeer(candidate peer.ID) {
+	if candidate != "" {
+		info := as.host.Peerstore().PeerInfo(candidate)
+		// TODO: track and exclude recently probed peers.
+		if !as.config.dialPolicy.skipPeer(info.Addrs) {
+			as.lastProbe = time.Now()
+			as.probe(&info)
+			return
+		}
```

Comment:
we probably don't want to probe anyways if we decide to skip this peer, right?

Reply:
this is probably named wrong. The intention is that it's a hint for a preferred peer to probe. If not specified, the function will try to find a 'nextPeer' on its own. perhaps "suggestedPeer"?